cartographer/mapping_3d/sparse_pose_graph.h

```diff
@@ -104,9 +104,9 @@ class SparsePoseGraph : public mapping::SparsePoseGraph {
   mapping::MapById<mapping::NodeId, mapping::TrajectoryNode>
   GetTrajectoryNodes() override EXCLUDES(mutex_);
   std::vector<Constraint> constraints() override EXCLUDES(mutex_);
-  void SetInitialTrajectoryPose(int from_trajectory_id, int to_trajectory_id,
-                                const transform::Rigid3d& pose,
-                                const common::Time time) override
+  void SetInitialTrajectoryPose(
```

Comment:
I am not sure why this is required. Adding the initial pose through the TrajectoryBuilderOptions should be sufficient, right?

Reply:
TrajectoryBuilderOptions is not available in sparse_pose_graph. Maybe you mean that I should do this https://github.com/googlecartographer/cartographer/pull/606#discussion_r149653974.